src/Raven.Server/Rachis/Leader.cs

```diff
@@ -677,13 +678,15 @@ private class RachisMergedCommand
 
         public async Task<(long Index, object Result)> PutAsync(CommandBase command, TimeSpan timeout)
         {
+            using var ctxResultWriter = command is IContextResultCommand crCommand ? new CtxResultWriter(crCommand.WriteResult) : null; 
```

Comment:
@haludi Ping me about this, I would to see if we can get rid of the `IContextResultCommand` completely.
My idea is to pass the result context to the `PutAsync`